platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/internal/WiringFlowchart.java

```diff
@@ -238,9 +243,17 @@ private void collapseGroup(@NonNull final GroupVertex group) {
         final List<ModelEdge> edges = collectEdges();
         final List<ModelVertex> groupVertices = collectGroupVertices(group);
 
+        final Set<String> hyperlinks = new HashSet<>();
+        for (final ModelVertex vertex : groupVertices) {
+            if (vertex.getHyperlink() != null) {
+                hyperlinks.add(vertex.getHyperlink());
+            }
+        }
+        final String hyperlink = hyperlinks.size() == 1 ? hyperlinks.iterator().next() : null;
```

Comment:
Minor: we go into the `else` branch if there's either zero or more than one element in the set (also, sets are unsorted, btw.) I'm not entirely sure if this is the desired logic.

If we're certain there's either zero or 1 element, then we could use something like `MoreCollectors.toOptional()` to enforce the contract and make the code robust.

If, on the other hand, more than 1 elements is allowed, then it's unclear why an unsorted set is used, and also why this is treated the same as if there's no elements at all. In this case, either the logic should be fixed, or some clarifying comments need to be added.

Reply:
This is intentional. When grouping things on a graph, the parent group should only have a hyperlink from a child if there is exactly one hyperlink within its set of children. If there are no child hyperlinks then there is obviously nothing we can do. But if there are multiple child hyperlinks, we have no way to decide which child hyperlink to choose. In the indeterminate case, better to have no hyperlink than to have a confusing and potentially misleading hyperlink.

Reply:
Makes sense. Thanks for clarifying this. However, like I mentioned, it would be best to add a comment about this into the code above this logic so that future maintainers of the code understand the intention.